2-mandatory/3-shopping-list.js

```diff
@@ -19,7 +19,12 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+  const newrecipe = { name: recipe.name, items: [] };
+  newrecipe.items = recipe.ingredients.filter(
+    (item) =>
+      !pantry.fridgeContents.concat(pantry.cupboardContents).includes(item)
+  );
```

Comment:
well done

Reply:
the `pantry.fridgeContents.concat(pantry.cupboardContents)` can be outside the filter since it needs to be done only once

Reply:
thank you